packages/node_modules/@node-red/runtime/lib/flows/Flow.js

```diff
@@ -161,14 +161,14 @@ class Flow {
             for (let i = 0; i < configNodes.length; i++) {
                 const node = this.flow.configs[configNodes[i]]
                 if (node.type === 'global-config' && node.env) {
-                    const nodeEnv = await flowUtil.evaluateEnvProperties(this, node.env, credentials.get(node.id))
+                    const nodeEnv = await flowUtil.evaluateEnvProperties(this, node.env, credentials.get(node.id) || {})
```

Comment:
Would it make sense to rather make the fallback explicit and required in the API itself, e.g. something like `credentials.getWithFallback(node.id, {})` so that it is impossible to forget?

The same fallback handling that is added here is already present in `packages/node_modules/@node-red/runtime/lib/flows/Subflow.js` however `credentials.get` is also called in `packages/node_modules/@node-red/runtime/lib/flows/Group.js` where the result is passed along to `flowUtil.evaluateEnvProperties` which does not have any checks and would unconditionally call `hasOwnProperty` on undefined when `credentials.get` returns undefined.

And in `packages/node_modules/@node-red/editor-api/lib/editor/index.js` the potential undefined result would be passed along as a callback to express as far as I can see and I assume that would be a problem.

In `packages/node_modules/@node-red/runtime/lib/nodes/index.js` the result is properly checked however it can be simplified by letting the initial `creds` assignment always having a valid value from calling `getWithFallback` (cloning and iterating over an empty object is just a no-op).

Reply:
@hlovdal thanks for the comments.

I briefly considered other approaches, such as having `credentials.get()` always return an empty object rather than null/undefined, but decided to focus the fix on the issue at hand rather than make broader changes.

Thanks for highlighting the Group object needed fixing as well - that made me reconsider the original fix. I have just pushed a much simpler one so that `flowUtil .evaluateEnvProperties` handles a null/undefined credentials object being passed in. That ensures it works for all cases and we won't miss it if we added another call to the function elsewhere.

> And in packages/node_modules/@node-red/editor-api/lib/editor/index.js the potential undefined result would be passed along as a callback to express as far as I can see and I assume that would be a problem.

No - that is an entirely different `credentials.get` and is correct in what it does.